### ISSUE TITLE
Add redirect parameters on logins on WCA main page

### ISSUE THREAD

**Issue by dunkOnIT** [thewca/worldcubeassociation.org#7380]
Redirect parameters should be present on every login. The absence of this makes the entire process a bit more cumbersome and potentially confusing for people who are not used to the platform.

For example, take this process:

- Click on a WCA statistics database query URL
- Login required
- User logs in
- Back to home screen. How do I get to the first screen again?

It is a bit counterintuitive. The same happens for any instance of login.

_Originally posted by @GuidoDipietro in https://github.com/thewca/worldcubeassociation.org/issues/6777#issuecomment-1311769349_
     
The above issue relates to the statistics page (see [this Issue](https://github.com/thewca/statistics/issues/92)), but the problem also exists on the WCA website, per Gregor's explanation: 

> It is important to point out the difference between OAuth login an on-premise login (i.e. casually browsing our website and clicking the "Login" button on the top right of the page)
> 
> OAuth has redirect URLs, and I am very confident that those are actually honored because we use a widely spread OAuth third party integration tool. If something about redirect URLs there was broken, the wider Ruby on Rails community would be affected and there would likely be a fast fix. In the example from the original post. it's really probably Alexandre's statistics website not setting the redirect URL correctly.
> 
> On premise logins use Devise user management, which does not set any redirect URL by default. The sign-in button is one of the very few exceptions, most likely it was a spontaneous soluton to a user complaint. But it is not a general feature of the wider website.

**Comment by gregorbg** [comment]
Relevant information for people who want to take a shot at implementing: https://github.com/heartcombo/devise/wiki/How-To:-%5BRedirect-back-to-current-page-after-sign-in,-sign-out,-sign-up,-update%5D

**Comment by FinnIckler** [comment]
Isn't this an issue with statistics not with the WCA oauth implementation?

**Comment by Randomno** [comment]
I believe the example is but in general, clicking the sign in button on the WCA page should include a redirect

**Comment by FinnIckler** [comment]
yes, but is it up to statistics to set to correct redirect URL. Otherwise redirects wouldn't work at all

**Comment by Randomno** [comment]
I was thinking outside the stats page though. For example you go to a competition page to accept registrations, realise you are not logged in, press Sign in, afterwards you are redirected back to the comp page.

**Comment by FinnIckler** [comment]
I just tested that and that works. When you login on a register page you will get redirected back to the register page after logging in

**Comment by gregorbg** [comment]
> I just tested that and that works. When you login on a register page you will get redirected back to the register page after logging in

Curious, it should not work like that.

**Comment by Randomno** [comment]
![Screenshot_20221115-130346_Firefox](https://user-images.githubusercontent.com/17330834/201926541-9985abe2-1de0-41ad-a0ab-65a20df97196.png)
This button?

Another bad example I guess. My first example (that I edited) was accepting registrations for a competition. In that instance you would want to be sent right back to the competition page. Is there any instance where it shouldn't redirect? Being sent back to the home page isn't very useful.

**Comment by gregorbg** [comment]
It is important to point out the difference between OAuth login an on-premise login (i.e. casually browsing our website and clicking the "Login" button on the top right of the page)

OAuth has redirect URLs, and I am very confident that those are actually honored because we use a widely spread OAuth third party integration tool. If something about redirect URLs there was broken, the wider Ruby on Rails community would be affected and there would likely be a fast fix. In the example from the original post. it's really probably Alexandre's statistics website not setting the redirect URL correctly.

On premise logins use Devise user management, which does not set any redirect URL by default. The sign-in button is one of the very few exceptions, most likely it was a spontaneous soluton to a user complaint. But it is not a general feature of the wider website.

**Comment by FinnIckler** [comment]
Then I think we should have two different issues. One is on the statistics repo about adding a redirect URL and one is to add a redirect to all the logins on the main page

**Comment by dunkOnIT** [comment]
[Issue](https://github.com/thewca/statistics/issues/92) created in the Statistics repo, and title/initial post of this Issue updated to reflect Finn's suggestion.

**Comment by dunkOnIT** [comment]
Since this, we have added a "redirect to last page visited" feature upon logging in, which has been very well received on the Register page in particular :)
